WeakAurasOptions/AuthorOptions.lua

```diff
@@ -1100,9 +1112,94 @@ typeControlAdders = {
+            local childData = optionData.data
+            if childOption.nameSource == -1 then
+              if value < childData.size then
+                for i = value + 1, childData.size do
+                  childOption.entryNames[i] = nil
+                end
+              else
+                for i = childData.size + 1, value do
+                  childOption.entryNames[i] = L["Entry %i"]:format(i)
+                end
+              end
+            end
+            childOption.size = value
+            WeakAuras.Add(childData)
+          end
+          WeakAuras.ReloadTriggerOptions(data)
+        end,
         disabled = function() return option.limitType == "none" end,
       }
+      local nameSources = CopyTable(WeakAuras.array_entry_name_types)
+      local validNameSourceTypes = WeakAuras.name_source_option_types
+      if option.limitType ~= "fixed" then
+        nameSources[-1] = nil
+      end
+      for subIndex, subOption in ipairs(option.subOptions) do
+        if validNameSourceTypes[option.type] then
+          local allShareThisOption = true
+          for id in pairs(option.references) do
+            if not subOption.references[id] then
+              allShareThisOption = false
+              break
+            end
+          end
+          if allShareThisOption then
+            nameSources[subIndex] = subOption.key
+          end
+        end
+      end
+      args[prefix .. "nameSource"] = {
+        type = "select",
+        name = name(option, "generateNames", L["Auto-Generate Entry Names"]),
+        order = order(),
+        values = nameSources,
+        width = WeakAuras.normalWidth,
+        get = function()
+          return option.nameSource or 0
+        end,
+        set = function(_, value)
+          for id, optionData in pairs(option.references) do
+            local childOption = optionData.options[optionData.index]
+            local childData = optionData.data
+            if (value == -1) ~= (childOption.nameSource == -1) then
+              if value == -1 then
+                local entryNames = {}
+                for i = 1, #childOption.subOptions do
+                  entryNames[i] = L["Entry #i"]:format(i)
+                end
+                childOption.entryNames = entryNames
+              else
+                childOption.entryNames = nil
+              end
+            end
+            if value > 0 then
+              childOption.nameSource = option.subOptions[value].references[id].optionIndex
+            else
+              childOption.nameSource = value
+            end
+            WeakAuras.Add(childData)
+          end
+          WeakAuras.ReloadTriggerOptions(data)
+        end,
+      }
+      if option.nameSource == -1 then
+        for i = 1, option.size do
+          args[prefix .. "entry" .. i .. "name"] = {
+            type = "input",
+            name = nameArray(option, "entryNames", i, L["Entry %i"]:format(i)),
```

Comment:
These helper functions aren't yet implemented, which will probably cause the CI check to fail.